whereby/browser-sdk
# Add "useLocalMedia" hook for local media handling

Expose a new hook to allow management of local media (audio, video).

### Tested like
1. `STORYBOOK_ROOM=<unlocked_room_in_production> yarn storybook`
2. Join your unlocked room as normal in one tab
3. Go to http://localhost:6006/?path=/story/examples-custom-ui--local-media-only
   - Switch between local cameras, verify the preview updates
   - Toggle camera on off, verify you get black screen
   - Navigate to the "Start / Stop" story to verify your camera lights are turned off
4. Go to http://localhost:6006/?path=/story/examples-custom-ui--room-connection-with-local-media
   - Choose camera and microphone, join room
   - Verify you get audio and video
   - Try using either toggle button to verify you are able to toggle cam / mic, and state is updated correctly for the remote participant
   - Try changing devices, verify the new device is being used in the call
   - Leave room by going to the "Start / Stop" story again, verify camera lights turn off

Todo:
- [x] Handle GUM error
- [x] Support toggling camera + microphone
- [x] Support enumerating devices
- [x] Support switching devices
